File: benchmarks/collatz.rb
def steps(start)
  n = start
  c = 0
  while n > 1
    if n % 2 == 0
      n = n / 2
    else
      n = 3 * n + 1
    end
    c += 1
  end
  c
end

t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
total = 0
i = 1
while i < 1000000
  total += steps(i)
  i += 1
end
t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)

puts total
puts "bench_ns=#{t1 - t0}"

File: benchmarks/collatz.js
function steps(start) {
    let n = start;
    let c = 0;
    while (n > 1) {
        if (n % 2 === 0) {
            n = Math.floor(n / 2);
        } else {
            n = 3 * n + 1;
        }
        c += 1;
    }
    return c;
}

const t0 = process.hrtime.bigint();
let total = 0;
for (let i = 1; i < 1000000; i++) {
    total += steps(i);
}
const t1 = process.hrtime.bigint();

console.log(total);
console.log(`bench_ns=${t1 - t0}`);
